doc/releases.md

```diff
@@ -1,5 +1,100 @@
 # Releases
 
+## Version 1.21
+
+### Version 1.21.0
+
+**June 25, 2025**
+
+This release add support for sample information on hover for rasterized/datashaded plots, adds dendrogram plot, and adds logarithmic support for Histogram.
+Along with these new features, this release includes a number of enhancements, bug fixes, and improved documentation.
+Many thanks to the new contributors @philipc2, @NolanBrb, and @sand-hanitizer, as well as the returning contributors
+@Azaya89, @LecrisUT, @maximlt, @michaelaye, @philippjfr, and @hoxbro.
+
+This release bumps the minimum supported Python version to 3.10, deprecates the
+`streamz` interface, autoloading RC file, IPython magic, and using matplotlib
+as default backend for `hv.extension` which are all planned for removal in
+1.23.0.
+
+New features:
+
+- Sample information on hover for rasterized/datashaded Bokeh plot (
+  [#6422](https://github.com/holoviz/holoviews/pull/6422),
+  [#6479](https://github.com/holoviz/holoviews/pull/6479),
+  [#6531](https://github.com/holoviz/holoviews/pull/6531),
+  [#6553](https://github.com/holoviz/holoviews/pull/6553),
+  [#6568](https://github.com/holoviz/holoviews/pull/6568),
+  [#6581](https://github.com/holoviz/holoviews/pull/6581),
+  [#6584](https://github.com/holoviz/holoviews/pull/6584),
+  [#6593](https://github.com/holoviz/holoviews/pull/6593),
+  [#6596](https://github.com/holoviz/holoviews/pull/6596),
+  [#6609](https://github.com/holoviz/holoviews/pull/6609))
+- Add dendrogram plot ([#6511](https://github.com/holoviz/holoviews/pull/6511),
+  [#6625](https://gthub.com/holoviz/holoviews/pull/6625),
+  [#6600](https://github.com/holoviz/holoviews/pull/6600))
+- Add logarithmic support for Histogram ([#6575](https://github.com/holoviz/holoviews/pull/6575))
+
+Breaking:
```

Comment:
You meant "Breaking change"? I'd just put this in the "Compatibility" section to be honest.

Reply:
I did. I was contemplating doing this, will do this.